### PR TITLE
Hatch 485: add enum doc

### PR DESCRIPTION
**Pull request by CMLabay**
# Description

# Jira ticket links

# Test Plan

# Definition of done

- [ ] Does new code have 90% testing coverage (100% for `core`) of both unit and E2E tests?
- [ ] Is code secure? If applicable, add security notes to the description.
- [ ] Do all new TODO comments have Jira links with enough information?
- [ ] Were all changes published and deployed to [the grid](https://github.com/bitovi/hatchify-grid-demo)?
- [ ] Has the browser error-console been reviewed to not contain new errors introduced by these code changes?
- [ ] The site looks “good” above 1000px width.
- [ ] The site looks “good” when the window height is small. No double scrollbars.
- [ ] Were the changes tested to ensure 508 compliance?

### CODE REVIEW

File: doc/schema/attribute-types/enum.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+```
+
+## Parameters
+
+| key          | description                                                                                      |   type    | optional |  default  |
+| ------------ | ------------------------------------------------------------------------------------------------ | :-------: | :------: | :-------: |
+| `default`    | The default value of the attribute. <br/> Example: `enumerate({default: "Pending", values:[...])`| `String`  |   Yes    |`undefined`|
+| `primary`    | If the attribute is a primary key. <br/> Example: `enumerate({primary: true, values:[...])`      | `Boolean` |   Yes    |  `false`  |
+| `values`     | The options of the enum <br/> Example: `enumerate({values: ["Pending", "Failed", "Completed"]})` | `Number`  |   No     |           |
```
Comment (arthurpankiewicz) [inline doc/schema/attribute-types/enum.md:20]: `type` here should be `Array` of type string? (or any?) idk what the backend supports exactly

File: doc/schema/attribute-types/enum.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    attributes: {
+      status: "Pending" // or null
+    }
+  }
+}
+```
+
+### Mutating Data
+
+When creating or updating an enumerate attribute, one of the defined values or `null` must be provided. Any other value will return a service error.
```
Comment (arthurpankiewicz) [inline doc/schema/attribute-types/enum.md:55]: or `null` if not required*

File: doc/schema/attribute-types/enum.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+}
+```
+
+## Grid Behavior
+
+The enum value in its entirety is shown as a string. Null values are shown as an empty table cell.
+
+![stringNullFilter](../../attachments/enum-grid.png)
```
Comment (arthurpankiewicz) [inline doc/schema/attribute-types/enum.md:76]: is `stringNullFilter` the fallback? if so it should be named differently

Comment (CMLabay) [reply doc/schema/attribute-types/enum.md:76]: I renamed that but I guess I didn't save or got `command z` happy 😅

File: doc/schema/attribute-types/enum.md (hunk 1 above)
Comment (CMLabay) [reply doc/schema/attribute-types/enum.md:20]: Thanks for catching that, it's definitely not number 😅

File: doc/schema/attribute-types/enum.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+The `enumerate` type will create a sequelize [DataTypes.ENUM](https://sequelize.org/docs/v6/other-topics/other-data-types/#enums) column.
+
+## Middleware Behavior
+
+### Querying Data
+
+```
+GET /todos?[status][$eq]=Pending  // all todos with status of Pending
+GET /todos?[status][$eq]=%00  // all todos with no status
+GET /todos?[status][$in][]=Pending&[status][$in][]Failed  // all todos with status of either Pending or Failed.
```
[the remark this answers is not in the document]
Comment (CMLabay) [reply doc/schema/attribute-types/enum.md:35]: It is, TY!

**Pull request merged**